compiler/modules/types/core.bal

```diff
@@ -815,9 +815,59 @@ function comparableNillableList(Context cx, SemType t1, SemType t2) returns bool
     }
     ComparableMemo memo = { semType1: t1, semType2: t2 };
     cx.comparableMemo.add(memo);
-    // SUBSET need to iterate members when tuples are supported
-    // following relies on the fact `listMemberType(cx, NIL, ()) = NEVER`
-    boolean result = comparable(cx, listMemberType(cx, t1, INT), listMemberType(cx, t2, INT));
+    boolean result;
+    var members1 = listAllMemberTypes(cx, t1);
+    var members2 = listAllMemberTypes(cx, t2);
+    // listAllMemberTypes return () if list type is not atomic
+    if members1 != () && members2 != () {
```

Comment:
This will be easier to understand if you factor out a function that merges the lists from the two calls to listAllMemberTypes into a `[Range,Semtype,Semtype][]`, then iterate over that.  Write a unit test for this merging function. Call something like mergeListMemberTypes.

Reply:
fixed